src/main/java/com/thoughtworks/pacman/ui/screens/ScorePanel.java

```diff
@@ -0,0 +1,125 @@
+package com.thoughtworks.pacman.ui.screens;
+
+import java.awt.*;
+import java.awt.event.*;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+import javax.swing.*;
+
+public class ScorePanel extends JFrame implements ActionListener {
+    JPanel panel;
+    JLabel user_label, message;
+    JTextField userName_text;
+    JButton submit;
+    String userName;
+    ArrayList<Entry> scoreList = new ArrayList<>();
+    int score;
+
+    public ScorePanel(int score) {
+        user_label = new JLabel();
+        user_label.setText("Your Score is High Enough to Join Top5! Enter Your Name");
+        userName_text = new JTextField();
+        submit = new JButton("Submit");
+        panel = new JPanel(new GridLayout(3, 1));
+        panel.add(user_label);
+        panel.add(userName_text);
+        message = new JLabel();
+        panel.add(submit);
+        setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
+        submit.addActionListener(this);
+        add(panel, BorderLayout.CENTER);
+        setTitle("Create new user");
+        setSize(350, 150);
+        setVisible(true);
+        this.score = score;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent ae) {
+        userName = userName_text.getText();
+        load();
+        Entry current = new Entry(userName, score);
+        scoreList.add(current);
+        sortList();
+        save();
+        setVisible(false);
+    }
+
+    public void load() {
+        String path = System.getProperty("user.dir");
```

Comment:
Bu kod ScoreBoard ta da tekrar ediyor. Demekki burada concernler karışmış durumda. Load implementasyonunu kendi klasına taşı ki burdan da o kodu çağırabilir. Böylece iki kere aynı kodu yazarak hem kod base şişmez hem de separation of concerns yaptığın için kodun bakımı/okunması/güncellenmesi/test edilmesi daha kolay olur